addons/account/models/account_move.py

```diff
@@ -797,7 +797,7 @@ def _get_pair_to_reconcile(self):
         elif self._context.get('skip_full_reconcile_check') == 'amount_currency_only':
             field = 'amount_residual_currency'
         #target the pair of move in self that are the oldest
-        sorted_moves = sorted(self, key=lambda a: a.date)
+        sorted_moves = sorted(self.with_context(prefetch_fields=False), key=lambda a: a.date)
```

Comment:
Because I forgot myself I ran some benchmarks again. These were conducted by registering a payment on ~1000 invoices with one invoice line each. Since most of the time is spent in auto_reconcile_lines I added some logging there to measure improvements in performance: https://gist.github.com/jorenvo/e098983276ad678e59d3378d05d6d92d. Here are my results: https://gist.github.com/jorenvo/d58cb5093f872b6232c574f3f11c0799. I honestly am not sure why the first patch helps. It generates a slighly faster query (I included it for completeness in the previous link). But that doesn't explain the performance improvement visible. Most likely there is something else going on.